cmake/modules/extensions.cmake

```diff
@@ -890,15 +893,17 @@ endfunction()
 #                   will be used as a valid revision for the board.
 #
 function(board_check_revision)
-  set(options EXACT)
+  set(options OPTIONAL EXACT)
   set(single_args FORMAT DEFAULT_REVISION HIGHEST_REVISION)
   set(multi_args  VALID_REVISIONS)
   cmake_parse_arguments(BOARD_REV "${options}" "${single_args}" "${multi_args}" ${ARGN})
 
   string(TOUPPER ${BOARD_REV_FORMAT} BOARD_REV_FORMAT)
 
   if(NOT DEFINED BOARD_REVISION)
-    if(DEFINED BOARD_REV_DEFAULT_REVISION)
+    if(BOARD_REV_OPTIONAL)
+      return()
```

Comment:
If `OPTIONAL` should be allowed as argument, then it must be made sure that it is mutual exclusive to the following parameters: `EXACT`, `DEFAULT_REVISION`

Reply:
Update: maybe `OPTIONAL` and `EXACT` doesn't need to be mutual exclusive, as `EXACT` can be considered to mean `EXACT` if a revision is given. 
But it would surely require an update to the function description to clearly indicate what `OPTIONAL` with `EXACT` means.